src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -187,7 +189,7 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                 };
                 subscribeToClientCertificateUpdatesNoCSR(getCertificateRequest, keyPair.getPublic(), consumer);
             }
-            metrics.subscribeSuccess();
+            domainEvent.emit(new MetricEvent(ClientDeviceAuthMetrics.METRIC_SUBSCRIBE_TO_CERTIFICATE_UPDATES_SUCCESS));
```

Comment:
This can work, but it also introduces coupling between the certificate and metric domains.

Ideally this code doesn't know anything about metrics at all. It should emit a certificate event and let an external listener translate that into a metric.

Perhaps you could add a CertificateSubscription event that includes details about the type of certificate and whether it succeeded or failed? Then we have flexibility to slice and dice the metrics however we want without needing to change this code.

Reply:
Is the type of certificate needed? Do we want to distinguish between server/client subscriptions or is one metric for both alright? I could always have the distinction there and only increment one metric in case we want to change it in the future though.

Reply:
IMO they should be separate metrics. Initially we may only want to emit metrics for server certificates since we don't expose client cert subscriptions to customers. But in the future we may, and tracking those separately is definitely something we would want